### PR TITLE
API workflows example

### LINKED ISSUE #7952: Write an example Go program for access requests
### What

Write an example go program that handles access requests and only uses our official api. Customers are looking at teleport-plugins repo that has dependency on lib/client and get stuck.

https://pkg.go.dev/github.com/gravitational/teleport/api/client

### PR DESCRIPTION
**Pull request by Joerger**
This PR access request example code using the official API client. The example was modeled after the original [access example](https://github.com/gravitational/teleport-plugins/tree/master/access/example).

Closes https://github.com/gravitational/teleport/issues/7952

### CODE REVIEW

File: examples/workflows/go.mod [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module example
```
Comment (andrejtokarcik) [inline examples/workflows/go.mod:1]: Doesn't matter probably but shouldn't this read `module workflows`?

File: examples/workflows/requester.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+kind: role
+metadata:
+  name: requester
+spec:
+  allow:
+    request:
+      roles:
+        - admin
+version: v3
```
Comment (andrejtokarcik) [inline examples/workflows/requester.yaml:9]: Shouldn't the roles have `version: v4` now?

File: examples/workflows/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+This example plugin demonstrates how to create a workflows plugin to automatically approve/deny new access requests based on a simple Allow List.
+
+## Demo
+
+This demo can be used to quickly get the example plugin up and running.
+
+##### Create resources
+
+```bash
```
Comment (andrejtokarcik) [inline examples/workflows/README.md:11]: I'm not sure `bash` is the correct highlighting mode for this and below "shell sessions" as these aren't actually bash scripts. Apparently `console` should be a better choice.

File: examples/workflows/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+## Example workflows plugin
+
+This example plugin demonstrates how to create a workflows plugin to automatically approve/deny new access requests based on a simple Allow List.
+
+## Demo
+
+This demo can be used to quickly get the example plugin up and running.
+
+##### Create resources
```
Comment (andrejtokarcik) [inline examples/workflows/README.md:9]:
```suggestion
### Create resources
```
here and below?

**Review by andrejtokarcik — APPROVED** [review]

File: examples/workflows/go.mod [hunk 5]
```diff
@@ -0,0 +1,11 @@
+module workflows
+
+go 1.16
+
+replace github.com/gravitational/teleport/api => ../../api
```
Comment (zmb3) [inline examples/workflows/go.mod:5]: Most people who follow this example won't be putting the code in-tree. Can we do without the replace directive here?

File: examples/workflows/main.go [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+	}
+	return conf, nil
+}
+
+func main() {
+	ctx := context.Background()
+
+	cfg, err := loadConfig("config.toml")
+	if err != nil {
+		log.Printf("ERROR: %s", err)
```
Comment (zmb3) [inline examples/workflows/main.go:68]: If you `log.Fatal()` there would be no need for the `os.Exit` call.

**Review by zmb3 — APPROVED** [review]
Couple minor comments but looks good.

**Review by klizhentas — APPROVED** [review]

**Pull request merged**